### PR TITLE
Remove hardcoded coredns

### PR DESCRIPTION
**Pull request by whites11**
This PR deletes the hardcoded coredns manifests from the terraform repo and uses the managed app instead

### CODE REVIEW

File: templates/files/conf/k8s-addons [hunk 1]
```diff
@@ -128,9 +128,17 @@ $KUBECTL -n kube-system delete deploy calico-kube-controllers --ignore-not-found
...
 $KUBECTL -n kube-system delete daemonset nginx-ingress-controller --ignore-not-found=true
 
+# delete coredns manifests
+$KUBECTL -n kube-system delete svc coredns --ignore-not-found=true
+$KUBECTL -n kube-system delete serviceaccount coredns --ignore-not-found=true
+$KUBECTL -n kube-system delete cm coredns --ignore-not-found=true
+$KUBECTL -n kube-system delete deploy coredns --ignore-not-found=true
+$KUBECTL delete clusterrolebinding system:coredns --ignore-not-found=true
+$KUBECTL delete clusterrole system:coredns --ignore-not-found=true
+
```
Comment (calvix) [inline templates/files/conf/k8s-addons:138]: what happens if this is rerun after the app is installed?

Comment (calvix) [reply templates/files/conf/k8s-addons:138]: maybe some if that will check if the app exists and delete only if it does not exists yet?

File: templates/files/k8s-resource/psp-bindings.yaml [hunk 2]
```diff
@@ -11,12 +11,6 @@ subjects:
   name: calico-kube-controllers
   namespace: kube-system
 {{- end }}
-- kind: ServiceAccount
-  name: coredns
-  namespace: kube-system
-- kind: ServiceAccount
-  name: nginx-ingress-controller
-  namespace: kube-system
```
[the remark this answers is not in the document]
Comment (whites11) [reply templates/files/k8s-resource/psp-bindings.yaml:19]: accident

[the remark this answers is not in the document]
Comment (whites11) [reply templates/files/k8s-resource/psp-bindings.yaml:19]: restored

File: templates/files/conf/k8s-addons (hunk 1 above)
Comment (whites11) [reply templates/files/conf/k8s-addons:138]: very good point. I thought of this while doing nginx ingress controller (PR coming), didn't check for coredns.
Checking

Comment (whites11) [reply templates/files/conf/k8s-addons:140]: how about this?

Comment (calvix) [reply templates/files/conf/k8s-addons:140]: looks good if it works as expected :p

**Review by calvix — APPROVED** [review]

**Pull request merged**